### ISSUE TITLE
How to contribute documentation to COBRApy?

### ISSUE THREAD

**Issue by natJimenez** [opencobra/cobrapy#617]
I don't know if this should be posted here since I already solved my issue, but I think it could be added to the documentation to prevent other people having the same problem. 

I tried to install cobrapy using pip directly into my computer (macOS Sierra) without virtualenv and I found that since the OS X capitan upgrade there have been some issues using pip (For more details see [How to use pip after the OS X El Capitan upgrade?](https://apple.stackexchange.com/questions/209572/how-to-use-pip-after-the-os-x-el-capitan-upgrade))

I manage to install cobrapy only for my user which is not a problem since I'm the only user on this computer, using
`pip install --user cobra`

Also I wanted to ask if there is a more direct way to contribute to your documentation? 

Thanks!

**Comment by Midnighter** [comment]
> Also I wanted to ask if there is a more direct way to contribute to your documentation?

Any and all contributions to the documentation are very welcome! It depends a little if you want to change the general documentation or function documentation since the latter is auto-generated. So the way forward is to follow the first part of the contributing instructions on [forking the repository](https://github.com/opencobra/cobrapy/blob/devel/.github/CONTRIBUTING.rst#get-started). Then change the specific `.rst` file in `documentation_builder` or modify the function docstrings directly. When you're satisfied you issue a pull request to cobrapy's devel branch.

Let me know about your questions here and we'll be happy to help.

**Comment by pstjohn** [comment]
To your specific problem, I'd discourage you from doing too much development using the system python that ships with your mac. ([this](https://github.com/MacPython/wiki/wiki/Which-Python) page has more details on why)

[homebrew](https://brew.sh/) and [anaconda](https://www.anaconda.com/download/#macos) (or [miniconda](https://conda.io/miniconda.html)) would probably give you fewer problems in the future 😄

**Comment by cdiener** [comment]
I agree that the install instructions are not easy to find. For instance right now they are inside https://github.com/opencobra/cobrapy/blob/devel/INSTALL.rst which for some reason is not part of the docs. A troubleshooting guide might be a good idea as well since we receive a lot of install problem questions on the mailing list as well...

**Comment by Midnighter** [comment]
Good idea. Additionally, we want to focus cobrapy development in the next sprint which hopefully means we can release cobrapy without the old solvers and thus end up with a vastly simplified installation, i.e., pure Python.

**Comment by natJimenez** [comment]
> To your specific problem, I'd discourage you from doing too much development using the system python that ships with your mac.

I usually don't do it but I needed to quickly install cobrapy to work while being completely disconnected from the outside world. I knew I was not suppose to do so but I didn't know why, so thanks for the link.

> A troubleshooting guide might be a good idea as well since we receive a lot of install problem questions on the mailing list as well...

Maybe we should add that to the install.rst in github? How is that connected to the "official documentation"?

**Comment by cdiener** [comment]
> Maybe we should add that to the install.rst in github? How is that connected to the "official documentation"?

Yes definitely. I think is should be moved there.
